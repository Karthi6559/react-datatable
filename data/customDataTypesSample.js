import React from "react";
import { TableCell } from "@material-ui/core";

const customDataTypesSample = [
  {
    dataType: "number",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "orange" }}>{cellVal}</div>
      </TableCell>
    )
  },
  {
    dataType: "text",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "purple" }}>{cellVal}</div>
      </TableCell>
    )
  },
  {
    dataType: "iban",
    component: cellVal => (
      <TableCell>
        <div style={{ color: "red" }}>{cellVal}</div>
      </TableCell>
    )
  }
];

export default customDataTypesSample;
