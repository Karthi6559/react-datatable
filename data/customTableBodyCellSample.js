import React from "react";
import { TableCell } from "@material-ui/core";
import { cellValPropType, columnPropType } from "../src/proptypes";

const customTableBodyCellSample = ({ cellVal, column }) => {
  let val;
  switch (column.dataType) {
    case "boolean":
      if (cellVal) {
        val = (
          <div
            className="data-boolean"
            style={{ color: "green", textAlign: "center" }}
          >
            Yes
          </div>
        );
      } else {
        val = <div style={{ color: "red", textAlign: "center" }}>No</div>;
      }
      break;
    default:
      val = (
        <div className="data" style={{ color: "blue" }}>
          {cellVal}
        </div>
      );
      break;
  }
  if (val) {
    return <TableCell>{val}</TableCell>;
  }
  return <TableCell className="no-data" />;
};

customTableBodyCellSample.propTypes = {
  cellVal: cellValPropType,
  column: columnPropType
};

export default customTableBodyCellSample;
