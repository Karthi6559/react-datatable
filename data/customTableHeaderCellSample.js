import React from "react";
import { columnPropType } from "../src/proptypes";

const customTableHeaderCellSample = ({ column }) => {
  switch (column.dataType) {
    case "number":
      return <div className="number">{column.label}</div>;
    case "text":
      return <div className="text">{column.label}</div>;
    case "boolean":
      return <div className="boolean">{column.label}</div>;
    case "date":
      return <div className="date">{column.label}</div>;
    default:
      return <div className="default">{column.label}</div>;
  }
};

customTableHeaderCellSample.propTypes = {
  column: columnPropType
};

export default customTableHeaderCellSample;
