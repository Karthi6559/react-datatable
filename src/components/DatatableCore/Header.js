import React, { Component } from "react";
import { connect } from "react-redux";
import { TableHead, TableRow, TableCell } from "@material-ui/core";
import { columnsPropType } from "../proptypes/proptypes";

class Header extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { columns } = this.props;
    return (
      <TableHead>
        <TableRow>
          {columns.map((column, id) => (
            <TableCell key={column.id} align={id === 0 ? "left" : "right"}>
              <div>{column.label}</div>
            </TableCell>
          ))}
        </TableRow>
      </TableHead>
    );
  }
}

Header.propTypes = {
  columns: columnsPropType
};

const mapStateToProps = state => {
  return {
    columns: state.datatableReducer.data.columns
  };
};

export default connect(mapStateToProps)(Header);
