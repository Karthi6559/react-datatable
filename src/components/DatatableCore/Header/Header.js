import React, { Component } from "react";
import { connect } from "react-redux";
import { TableHead } from "@material-ui/core";
import {
  columnsPropType,
  columnsOrderPropType
} from "../../proptypes/proptypes";
import HeaderRow from "./HeaderRow";

class Header extends Component {
  headerRowBuilder = () => {
    const { columnsOrder, CustomTableHeaderRow } = this.props;

    if (CustomTableHeaderRow !== null) {
      return <CustomTableHeaderRow columnsOrder={columnsOrder} />;
    }
    return <HeaderRow />;
  };

  render() {
    return <TableHead>{this.headerRowBuilder()}</TableHead>;
  }
}

Header.propTypes = {
  columnsOrder: columnsOrderPropType
};

const mapStateToProps = state => {
  return {
    columnsOrder:
      state.datatableReducer.features.userConfiguration.columnsOrder,
    CustomTableHeaderRow: state.customComponentsReducer.CustomTableHeaderRow
  };
};

export default connect(mapStateToProps)(Header);
