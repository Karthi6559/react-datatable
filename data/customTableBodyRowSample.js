import React from "react";
import {
  rowPropType,
  columnsOrderPropType,
  indexPropType,
  heightNumberPropType,
  columnSizeMultiplierPropType
} from "../src/proptypes";
import { simpleOptionsSample } from "./samples";

const customTableBodyRowSample = ({
  row,
  columnsOrder,
  rowIndex,
  columnSizeMultiplier,
  height
}) => {
  const { columns } = simpleOptionsSample.data;
  const columnAction = {
    id: "actions",
    label: "Actions",
    colSize: "150px",
    editable: false
  };
  return (
    <div
      className={
        rowIndex % 2 === 0 ? "Table-Row stripped" : "Table-Row not-stripped"
      }
      style={{
        height,
        background: rowIndex % 2 === 0 ? "#3f3f3f" : "#5f5f5f"
      }}
    >
      {columnsOrder.map(columnId => {
        const column =
          columnId === "actions"
            ? columnAction
            : columns.find(col => col.id === columnId);
        const width = `${(
          column.colSize.split("px")[0] * columnSizeMultiplier
        ).toString()}px`;
        return (
          <div className="Table-Cell" key={columnId}>
            <div style={{ color: "pink", width }}>{row[columnId]}</div>
          </div>
        );
      })}
    </div>
  );
};

customTableBodyRowSample.propTypes = {
  row: rowPropType,
  columnsOrder: columnsOrderPropType,
  rowIndex: indexPropType,
  height: heightNumberPropType,
  columnSizeMultiplier: columnSizeMultiplierPropType
};

export default customTableBodyRowSample;
