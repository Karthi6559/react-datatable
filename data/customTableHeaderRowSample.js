import React from "react";
import {
  columnsOrderPropType,
  columnSizeMultiplierPropType
} from "../src/proptypes";
import { simpleOptionsSample } from "./samples";

const customTableHeaderRowSample = ({ columnsOrder, columnSizeMultiplier }) => {
  const { columns } = simpleOptionsSample.data;
  const columnAction = {
    id: "o2xpActions",
    label: "Actions",
    colSize: "150px",
    editable: false
  };
  return (
    <div className="Table-Row">
      {columnsOrder.map(columnId => {
        const column =
          columnId === "o2xpActions"
            ? columnAction
            : columns.find(col => col.id === columnId);
        const width = `${(
          column.colSize.split("px")[0] * columnSizeMultiplier
        ).toString()}px`;
        return (
          <div className="Table-Header-Cell" key={column.id}>
            <div style={{ color: "green", width }}>{column.label}</div>
          </div>
        );
      })}
    </div>
  );
};

customTableHeaderRowSample.propTypes = {
  columnsOrder: columnsOrderPropType,
  columnSizeMultiplier: columnSizeMultiplierPropType
};

export default customTableHeaderRowSample;
