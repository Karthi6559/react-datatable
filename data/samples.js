import storeSample from "./storeSample";
import storeNoCustomComponentsSample from "./storeNoCustomComponentsSample";
import storeCustomTableBodyCellComponentSample from "./storeCustomTableBodyCellComponentSample";
import storeCustomTableBodyRowComponentSample from "./storeCustomTableBodyRowComponentSample";
import storeCustomTableHeaderCellComponentSample from "./storeCustomTableHeaderCellComponentSample";
import storeCustomTableHeaderRowComponentSample from "./storeCustomTableHeaderRowComponentSample";
import storeNoDataSample from "./storeNoDataSample";
import storeNoRowsDataSample from "./storeNoRowsDataSample";
import simpleOptionsSample from "./simpleOptionsSample";
import simpleOptionsNoDataSample from "./simpleOptionsNoDataSample";
import maximumOptionsSample from "./maximumOptionsSample";
import minimumOptionsSample from "./minimumOptionsSample";
import defaultOptionsSample from "./defaultOptionsSample";
import mergedSimpleOptionsSample from "./mergedSimpleOptionsSample";
import mergedMaximumOptionsSample from "./mergedMaximumOptionsSample";
import mergedMinimumOptionsSample from "./mergedMinimumOptionsSample";
import customTableBodyRowSample from "./customTableBodyRowSample";
import customTableBodyCellSample from "./customTableBodyCellSample";
import customTableHeaderRowSample from "./customTableHeaderRowSample";
import customTableHeaderCellSample from "./customTableHeaderCellSample";
import customDataTypesSample from "./customDataTypesSample";

export {
  storeSample,
  storeNoCustomComponentsSample,
  storeCustomTableBodyCellComponentSample,
  storeCustomTableBodyRowComponentSample,
  storeCustomTableHeaderCellComponentSample,
  storeCustomTableHeaderRowComponentSample,
  storeNoDataSample,
  storeNoRowsDataSample,
  simpleOptionsSample,
  simpleOptionsNoDataSample,
  maximumOptionsSample,
  minimumOptionsSample,
  defaultOptionsSample,
  mergedSimpleOptionsSample,
  mergedMaximumOptionsSample,
  mergedMinimumOptionsSample,
  customTableBodyRowSample,
  customTableBodyCellSample,
  customTableHeaderRowSample,
  customTableHeaderCellSample,
  customDataTypesSample
};
