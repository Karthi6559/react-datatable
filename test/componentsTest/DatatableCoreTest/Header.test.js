import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Header from "../../../src/components/DatatableCore/Header";
import { storeSample } from "../../data/samples";

describe("Header component", () => {
  const mockStore = configureStore();
  beforeEach(() => {});
  it("should render without errors", () => {
    const store = mockStore(storeSample);
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
