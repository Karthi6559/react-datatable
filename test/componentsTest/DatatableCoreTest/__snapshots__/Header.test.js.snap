// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render without errors 1`] = `
<thead
  className="MuiTableHead-root-1"
>
  <tr
    className="MuiTableRow-root-2 MuiTableRow-head-5"
  >
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignLeft-15"
      scope="col"
    >
      <div>
        id
      </div>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <div>
        name
      </div>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <div>
        age
      </div>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <div>
        adult
      </div>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root-7 MuiTableCell-head-8 MuiTableCell-alignRight-17"
      scope="col"
    >
      <div>
        birth date
      </div>
    </th>
  </tr>
</thead>
`;
